### PR TITLE
Support for setUserId

### PR DESCRIPTION
**Pull request by mahavirvataliya**
Added Support for setUser Id

### CODE REVIEW

File: ios/RNMatomoSdk/RNMatomoSdk.swift [hunk 1]
```diff
@@ -54,4 +54,13 @@ class RNMatomoSdk: NSObject {
             rejecter("not_initialize", "The tracker has not been initialized", NSError())
         }
     }
+
+    @objc(setUserId:)
+    func setUserId(
+        userID: String
```
Comment (aeirola) [inline ios/RNMatomoSdk/RNMatomoSdk.swift]: For consistency, this could be `userId`

File: index.d.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ declare module 'react-native-matomo-sdk' {
     export function initialize(apiUrl: string, siteId: number): Promise<void>;
     export function trackView(route: string[]): Promise<void>;
     export function trackEvent(category: string, action: string, name?: string, value?: number): Promise<void>;
+    export function setUserId(userId: string): void;
```
Comment (aeirola) [inline index.d.ts]: I think it would be useful if the function returned a promise to let the caller know when the userId was successfully set.

**Review by aeirola — COMMENTED** [review]
Thank you for the clean and useful PR! If you have the time, I would appreciate it if you could add the promise return value to the `setUserId` function so that callers can know when the value has been set.

File: index.d.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ declare module 'react-native-matomo-sdk' {
     export function initialize(apiUrl: string, siteId: number): Promise<void>;
     export function trackView(route: string[]): Promise<void>;
     export function trackEvent(category: string, action: string, name?: string, value?: number): Promise<void>;
+    export function setUserId(userId: string): Promise<void>;
```
Comment (aeirola) [inline index.d.ts:6]: I think it should be possible to remove the userId as well. At least the Android SDK does this by passing a `null` value as the userId (https://github.com/matomo-org/matomo-sdk-android/blob/v4.1.2/tracker/src/main/java/org/matomo/sdk/Tracker.java#L332). Not sure how the iOS SDK handles it though.

**Review by aeirola — CHANGES_REQUESTED** [review]
Looks good, but seems to be missing the ability to unset the user id. Would you care to add that to the PR?

**Comment by aeirola** [comment]
@mahavirvataliya Did you have plans on making changes to support unsetting the user ID? If not, I'll probably just merge this PR and just add support for unsetting the user id separately.

**Comment by mahavirvataliya** [comment]
Right now i don't have plan but i will do it later

File: index.d.ts (hunk 3 above)
Comment (jozan) [reply index.d.ts:6]: iOS SDK works the same: https://github.com/matomo-org/matomo-sdk-ios/blob/develop/MatomoTracker/MatomoTracker.swift#L22

**Review by aeirola — APPROVED** [review]
It's OK if you don't have time to handle the clearing of the user ID. I'll merge this as it is, and fix the issue in a separate PR.

**Comment by aeirola** [comment]
This feature is now released in version 0.4.0!

**Pull request merged**